nilearn/datasets/tests/test_utils.py

```diff
@@ -433,13 +432,42 @@ def test_safe_extract(tmp_path):
 
 def test_fetch_file_part(tmp_path):
     url = "http://foo/temp.txt"
-    (tmp_path / "temp.txt.part").touch()
+    file_full = tmp_path / "temp.txt"
+    file_part = tmp_path / "temp.txt.part"
+    file_part.touch()
 
     _utils.fetch_single_file(
-        url=url, data_dir=str(tmp_path), verbose=0, resume=True
+        url=url, data_dir=tmp_path, verbose=0, resume=True
     )
 
-    assert (tmp_path / "temp.txt").exists()
+    assert file_full.exists()
+
+    file_full.unlink()
+    assert not file_full.exists()
+    assert not file_part.exists()
+
+    # test for overwrite
+    url = "http://foo/temp.txt"
+    file_part.touch()
+
+    _utils.fetch_single_file(
+        url=url, data_dir=tmp_path, verbose=0, resume=True, overwrite=True
+    )
+
+    assert file_full.exists()
+
+
+def test_fetch_file_non_existing_dir(tmp_path, request_mocker):
```

Comment:
Can you include a one-liner to explain what is tested here ?

Reply:
This test is not necessary. I added it when codecov complained about untested lines. Those lines are refactored afterwards.  I'll remove it.